packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -620,13 +620,25 @@ abstract class ExternalApplicationListener extends Resource implements IApplicat
    *
    * It's possible to add conditions to the TargetGroups added in this way.
    * At least one TargetGroup must be added without conditions.
+   *
+   * NOTE: if you are deploying new infrastructure using this method,
+   * it is recommended to set the `removeRuleSuffixFromLogicalId: false`
+   * property so that the logicalId for the generated `ListenerRule`
+   * is consistent with other methods for managing `ListenerRule`s.
```

Comment:
Sort of, in this case we don't want users to have to think about the existence of this prop at all. Any description of it should focus just on the migration use case. We don't have any reason to assume that all users will want to migrate in the future so this feels like cognitive overload.

Reply:
I kind of agree. Maybe addressing changes to the `addTargetGroups()` method should be a separate PR?